src/Evolution/DgSubcell/Tags/Coordinates.hpp

```diff
@@ -0,0 +1,45 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
+#include "Utilities/GetOutput.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim>
+class Mesh;
+namespace gsl {
+template <typename>
+struct not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace evolution::dg::subcell::Tags {
+/// The coordinates in a given frame.
+template <size_t Dim, typename Frame>
+struct Coordinates : db::SimpleTag {
+  static std::string name() noexcept {
+    return get_output(Frame{}) + "Coordinates";
```

Comment:
Prefixing this with `Subcell` would break visualizations, and I can't really spend time overhauling all the visualization scripts to support subcell at that level

Reply:
I'm fine leaving this as-is for now to avoid refactoring anything. Though I think it would be quite straight-forward: Since you have two different tags for the subcell/dg coordinates anyway you have to switch between them in your observation event, at which point you can just pick a consistent label for the dataset in the output file.

Question: Will the subcell code actually write into the same volume data as the DG code?

Reply:
No, not nearly that straightforward. I have a wrapper event that calls into the impl of the generic fields event. It is that event that decides. I'd have to do some refactoring of the generic event, then the subcell event, and then I'd have to refactor `Render1D`, and also `GenerateXdmf`. It's a pretty big project.

Yes, DG and subcell data are written into the same subfile. Writing into different subfiles will require an even bigger rewrite of the visualization scripts. I currently see no benefit to writing into different subfiles and lots of drawbacks.